src/Features/CSharp/Portable/RemoveUnnecessaryImports/AbstractCSharpRemoveUnnecessaryImportsService.cs

```diff
@@ -19,16 +19,18 @@ namespace Microsoft.CodeAnalysis.CSharp.RemoveUnnecessaryImports
     internal partial class AbstractCSharpRemoveUnnecessaryImportsService :
         AbstractRemoveUnnecessaryImportsService<UsingDirectiveSyntax>
     {
-        public override async Task<Document> RemoveUnnecessaryImportsAsync(
+        protected override async Task<Document> RemoveUnnecessaryImportsAsync(
             Document document, 
             Func<SyntaxNode, bool> predicate,
+            bool fromAllContext,
```

Comment:
what does this mean?

Reply:
The service would try to remove import only if it's deemed unnecessary in all contexts, i.e. if the document is linked or contained in a multi-TFM project, and we changed the code in this document that causes some imports to be unnecessary, but not yet made the same change in other documents linked to this document, the unnecessary import will not be removed as a result. 

The change I made to the remove imports service make it possible to ignore other contexts.

Reply:
> The service would try to remove import only if it's deemed unnecessary in all contexts, i.e. if the document is linked or contained in a multi-TFM project, 

Why would this not be what we want *all* the time? Why would you want to opt-out of this?

Reply:
This is no longer needed, so I reverted it.

In case anyone's interested, there were a couple reasons behind this change:
1. I wasn't handling all linked documents in multi-targeting projects, so the imports to be removed in one context (i.e. one of many linked documents) isn't unnecessary in other contexts.
2. Even after I made the change to properly handle MTFM projects, this was still needed because, the imports to be removed are added to the documents as part of the refactoring, and try to remove unnecessary ones from all documents will fail for all but first document because once the import is removed from the first document, it's no longer "common" among all documents, i.e. it's not even in the first document! As a result of this, the diff merge will keep the unnecessary import even though it is removed from one of the documents.

I have avoided this issue by using `LinkedFileDiffMergingSession` directly, twice. (Thanks to @dpoeschl for the tip 👍 )